### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by kalavhan**
pull request for milestone-3

### CODE REVIEW

**Review by imhta — COMMENTED** [review]
PR is under review by **Pon Muthu Selvam N**

**Review by imhta — CHANGES_REQUESTED** [review]
# Changes required 🛑 

### Hello @kalavhan @fegzycole, 👋 There are some issues to fix

- [x] The application should determine  if move is a winning move
- [x]  The application should detect a sequence of three horizontal elements that are equal
- [x]  The application should detect a sequence of three vertical elements that are equal
- [x]  The application should detect a sequence of three equal elements in a diagonal from left and right to rig

**And some Implementation details**
* [x] The project uses at least two classes (Usually it uses three: Game, Board and Player). The code of the project should be logically divided between classes. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/classes.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
* [x] The code should respect the single responsibility rule for methods (i.e. each method does exactly one thing). Student's tends to create long methods that are doing multiple steps of the game process. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/single_responsibility.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
* [x] All classes are kept in `lib` directory. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
* [x] Use attr_readers/attr_writers instead of attr_accessors. The student must be aware that by using it they allow writing that property from outside the class. That variable can be set to anything. <TODO: explain why> ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/attr_accessor.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
* [x] DRY rule should be respected. E.g. students tend to check winning position for X and O separately instead of creating a method and reuse it. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/dry.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

----
#### Don't forgot to request for another code review
----

**Pon Muthu Selvam N**
_If you have any queries, feel free to ping me on slack_

**Comment by fegzycole** [comment]
Hello @imhta 

Thank you for your review, However, we have done all of what was highlighted initially.

Regards

**Review by imhta — COMMENTED** [review]
I think, I mistakely check the wrong feature branch

#### Sorry :(

> Pon Muthu Selvam

File: bin/main.rb [hunk 1]
```diff
@@ -1,64 +1,44 @@
 #!/usr/bin/env ruby
 
-# Array of valid positions that a player can take
-valid_options = %w[1 2 3 4 5 6 7 8 9]
+require './lib/board'
+require './lib/validation'
```
Comment (imhta) [inline bin/main.rb]: Change this to `require_relative` and folder location is wrong, fix this before merge the project

**Review by imhta — APPROVED** [review]
# Project approved 🎉 

### Hello  @fegzycole  @kalavhan  , 👋 You have done a good job in this project 👏 



> Fix that inline comment before merge the project
----
### Happy coding 😄 
----
**Pon Muthu Selvam N**
__nponmuthuselvam@gmail.com__

**Pull request merged**